tools/cassandra_delete_range/main.go

```diff
@@ -277,94 +316,108 @@ func deleteLedgerData(cluster *gocql.ClusterConfig, fromLedgerIdx uint64, toLedg
 	log.Printf("Start scanning and removing data for %d -> latest (%d according to ledger_range table)\n\n", fromLedgerIdx, toLedgerIdx)
 
 	// successor queries
-	log.Println("Generating delete queries for successor table")
-	info, rowsCount, errCount = prepareDeleteQueries(cluster, fromLedgerIdx,
-		"SELECT key, seq FROM successor WHERE token(key) >= %s AND token(key) <= %s",
-		"DELETE FROM successor WHERE key = ? AND seq = ?")
-	log.Printf("Total delete queries: %d\n", len(info.Data))
-	log.Printf("Total traversed rows: %d\n\n", rowsCount)
-	totalErrors += errCount
-	totalRows += rowsCount
-	deleteCount, errCount = performDeleteQueries(cluster, &info, columnSettings{UseBlob: true, UseSeq: true})
-	totalErrors += errCount
-	totalDeletes += deleteCount
-
-	// diff queries
-	log.Println("Generating delete queries for diff table")
-	info, rowsCount, errCount = prepareDeleteQueries(cluster, fromLedgerIdx,
-		"SELECT key, seq FROM diff WHERE token(seq) >= %s AND token(seq) <= %s",
-		"DELETE FROM diff WHERE key = ? AND seq = ?")
-	log.Printf("Total delete queries: %d\n", len(info.Data))
-	log.Printf("Total traversed rows: %d\n\n", rowsCount)
-	totalErrors += errCount
-	totalRows += rowsCount
-	deleteCount, errCount = performDeleteQueries(cluster, &info, columnSettings{UseBlob: true, UseSeq: true})
-	totalErrors += errCount
-	totalDeletes += deleteCount
+	if !*skipSuccessorTable {
```

Comment:
If run time is important, then it should be possible to run each table in it's own go-routine reporting the metrics via a channel with a struct such as the one below:
```
struct OperationResult {
errCount int64
rowsAffected int64
deleteCount int64
}
```

Reply:
Sounds like a nice idea for another day 👍 🙇